src/Model/Config/Communication.php

```diff
@@ -123,4 +102,49 @@ protected function getChannel(string $langAbbr): string
 
         return $channels[$langAbbr];
     }
+
+    private function prepareParams(array $params, Category $category = null): array
+    {
+        $params['add-params'] = $this->useForCategories() ? $this->getStandardCategoryPath($category) : 'cl=search_result';
+
+        if ($this->getConfig('ffApiVersion') === 'ng') {
+            $params['category-page'] = $category ?
+                $this->useForCategories() ? $this->getNgCategoryPath($category) : null
+                : null;
+            $params['add-params']    = null;
+        }
+
+        return $params;
+    }
+
+    private function getNgCategoryPath(Category $category, string $param = 'CategoryPath'): string
```

Comment:
the optional argument value after change does not have much sense as the private method is not reachable from the child class in case user wants to override it.
Previously the method was protected which had a sense as user might have overridden only this method to change the $param argument value. After this PR user would need to override the `getParameters`
as the `prepareAddParams` is also private